src/app/shared/components/template/components/task-card/task-card.component.ts

```diff
@@ -1,16 +1,91 @@
 import { Component, OnInit } from "@angular/core";
 import { TaskService } from "src/app/shared/services/task/task.service";
-import { getStringParamFromTemplateRow } from "src/app/shared/utils";
+import {
+  getStringParamFromTemplateRow,
+  getVariantParamFromTemplateRow,
+} from "src/app/shared/utils";
 import { TemplateBaseComponent } from "../base";
 import { IProgressStatus } from "src/app/shared/services/task/task.service";
 import { TemplateFieldService } from "../../services/template-field.service";
 
+interface ITaskCardParams {
+  /**
+   * The ID of an individual task. A task card can display *either* an individual task,
+   * *or* a task group: either a task_id or a task_group_id should be provided.
+   */
+  task_id: string;
+  /**
+   * The ID of a task group, used to track whether it is the highlighted task group.
+   * If provided, the task card will display a progress bar for the task group
+   * **/
+  task_group_id: string;
+  /**
+   * The name of the data_list containing the elements of the task group,
+   * used to track completion progress of the task group
+   * **/
+  task_group_data: string;
+  /**
+   * The name of the field that tracks the completion of the task. Can be omitted when displaying a task group,
+   * as their completion is calculated by the completion of their constituent tasks
+   */
+  completed_field: string;
+  /** The title to display on the task card */
+  title: string;
+  /**
+   * The subtitle to display on the task card, typically a short description.
+   * Optional
+   */
+  subtitle: string;
+  /**
+   * The image to display on the task card.
+   * Optional
+   */
+  image: string;
+  /** Legacy, use variant instead */
+  style: string;
+  /**
+   * A list of named style variants of the component, separated by spaces or commas.
+   * Use interchangeably with "variant_list". Default "landscape"
+   * */
+  variant: string;
+  /**
+   * A list of named style variants of the component, separated by spaces or commas.
+   * Use interchangeably with "variant". Default "landscape"
+   * */
+  variant_list: string;
```

Comment:
nit(blocking)
Can't remember exactly what we decided on the call, but I can't really think of any reason why we need to support list format for variants, as all variants are simple strings so a list really still just a string like `portrait background-500`. The only potential would be loading a whole list of styles from a data_list, however this feels unnecessarily complex to me so recommend removing all list references